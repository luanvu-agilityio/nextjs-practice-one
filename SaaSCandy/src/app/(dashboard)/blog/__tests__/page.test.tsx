import { render } from '@testing-library/react';
import Blog from '@/app/(dashboard)/blog/page';

jest.mock('@/components/layout/PageLayout', () => {
  return function PageLayout({
    title,
    subtitle,
    children,
  }: {
    title: string;
    subtitle: string;
    children: React.ReactNode;
  }) {
    return (
      <div data-testid='page-layout'>
        <h1>{title}</h1>
        <p>{subtitle}</p>
        {children}
      </div>
    );
  };
});

jest.mock('@/components/BlogPage', () => {
  return function BlogPage() {
    return <div data-testid='blog-page'>Blog Page Content</div>;
  };
});

describe('Blog Page - Snapshot Tests', () => {
  it('should match snapshot', () => {
    const { container } = render(<Blog />);
    expect(container).toMatchSnapshot();
  });

  it('should render with correct title and subtitle', () => {
    const { getByText } = render(<Blog />);

    expect(getByText('Blog')).toBeInTheDocument();
    expect(
      getByText(/Discover a wealth of insightful materials/)
    ).toBeInTheDocument();
  });

  it('should render BlogPage component', () => {
    const { getByTestId } = render(<Blog />);

    expect(getByTestId('blog-page')).toBeInTheDocument();
  });
});
