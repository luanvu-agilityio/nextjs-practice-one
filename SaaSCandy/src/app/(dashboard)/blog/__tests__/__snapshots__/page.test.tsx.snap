// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Blog Page - Snapshot Tests should match snapshot 1`] = `
<div>
  <div
    data-testid="page-layout"
  >
    <h1>
      Blog
    </h1>
    <p>
      Discover a wealth of insightful materials meticulously crafted to provide you with a comprehensive understanding of the latest trends.
    </p>
    <div
      data-testid="blog-page"
    >
      Blog Page Content
    </div>
  </div>
</div>
`;
